install_config/persistent_storage/persistent-volume-snapshots.adoc

```diff
@@ -0,0 +1,401 @@
+* Supported `PersistentVolume` link:https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes[types]:
+** AWS Elastic Block Store (EBS) 
+** Google Compute Engine (GCE) Persistent Disk (PD)
+
+[[installation-setup]]
+== Installation and Setup
+
+The external controller and provisioner are the external components that provide
+volume snapshotting. These external components run in the cluster. The
+controller is responsible for creating, deleting, and reporting events on volume
+snapshots. The provisioner creates new `PersistentVolumes` from the volume
+snapshots. See xref:persistent-volume-snapshots#create-snapshot[Create Snapshot]
+and xref:persistent-volume-snapshots#restore-snapshot[Restore Snapshot] for
+more information.
+
+[[start-controller-and-provisioner]]
+=== Starting the External Controller and Provisioner
+
+The external controller and provisioner services are distributed as container
+images and can be run in the [product-title] cluster as usual. To allow the
+containers managing the API objects, the necessary role-based access control
+(RBAC) rules need to be configured by the administrator:
+
+. Create a `ServiceAccount` and `ClusterRole`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: snapshot-controller-runner
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: snapshot-controller-role
+rules:
+  - apiGroups: [""]
+    resources: ["persistentvolumes"]
+    verbs: ["get", "list", "watch", "create", "delete"]
+  - apiGroups: [""]
+    resources: ["persistentvolumeclaims"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["storageclasses"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+  - apiGroups: ["apiextensions.k8s.io"]
+    resources: ["customresourcedefinitions"]
+    verbs: ["create", "list", "watch", "delete"]
+  - apiGroups: ["volumesnapshot.external-storage.k8s.io"]
+    resources: ["volumesnapshots"]
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+  - apiGroups: ["volumesnapshot.external-storage.k8s.io"]
+    resources: ["volumesnapshotdatas"]
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+----
+
+. Bind the rules via `ClusterRoleBinding`:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: snapsot-controller
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: snapshot-controller-role
+subjects:
+- kind: ServiceAccount
+  name: snapshot-controller-runner
+  namespace: default
+----
+
+If the external controller and provisioner are deployed in Amazon Web Services
+(AWS), they must be able to authenticate using the access key. To provide the
+credential to the pod, the administrator creates a new secret:
```

Comment:
These paragraphs and codeblocks that follow are separate from the procedure above it, right?

Reply:
Yes.